hat/assets/js/apps/Iaso/domains/app/components/LanguageSwitchComponent.js

```diff
@@ -21,19 +21,22 @@ const useStyles = makeStyles(theme => ({
 export default function LanguageSwitchComponent() {
     const classes = useStyles();
     const [anchorEl, setAnchorEl] = React.useState(null);
-    const activeLocale = useSelector(state => state.app.locale);
-    const dispatch = useDispatch();
+    const { locale: activeLocaleCode, setLocale } = useLocale();
+    const activeLocale = APP_LOCALES.find(
+        locale => locale.code === activeLocaleCode,
+    );
+
+    const { mutate: saveCurrentUser } = useSaveCurrentUser(false);
 
     const handleClickListItem = event => {
         setAnchorEl(event.currentTarget);
     };
 
     const handleLocaleSwitch = localeCode => {
-        dispatch(
-            saveCurrentUserProFile({
-                language: localeCode,
-            }),
-        );
+        setLocale(localeCode);
+        saveCurrentUser({
+            language: localeCode,
+        });
```

Comment:
I don't understand why this works: `saveCurrentUser` doesn't invalidate the data in react-query's cache, and `useCurrentUser` accesses the data directly, without refetching if it's stale like a `useQuery` hook would.

My instinct would be to :
- either change `saveCurrentUser` to invalidate the cache
- or add an `onSuccess` option to `saveCurrentUser` to overwrite the cache data with the API response.

But if works better like this, well 🤷‍♂️